src/qml/nodemodel.cpp

```diff
@@ -68,10 +78,16 @@ void NodeModel::timerEvent(QTimerEvent* event)
 void NodeModel::ConnectToBlockTipSignal()
 {
     assert(!m_handler_notify_block_tip);
+    assert(!m_handler_notify_num_peers_changed);
 
     m_handler_notify_block_tip = m_node.handleNotifyBlockTip(
         [this](SynchronizationState state, interfaces::BlockTip tip, double verification_progress) {
             setBlockTipHeight(tip.block_height);
             setVerificationProgress(verification_progress);
         });
+
+    m_handler_notify_num_peers_changed = m_node.handleNotifyNumConnectionsChanged(
```

Comment:
why lump this in here, isn't the following cleaner?

``` diff
diff --git a/src/qml/nodemodel.cpp b/src/qml/nodemodel.cpp
index 533baf5ef..3e2232c8a 100644
--- a/src/qml/nodemodel.cpp
+++ b/src/qml/nodemodel.cpp
@@ -18,6 +18,7 @@ NodeModel::NodeModel(interfaces::Node& node)
     : m_node{node}
 {
     ConnectToBlockTipSignal();
+    ConnectToNumConnectionsChangedSignal();
 }
 
 void NodeModel::setBlockTipHeight(int new_height)
@@ -78,16 +79,20 @@ void NodeModel::timerEvent(QTimerEvent* event)
 void NodeModel::ConnectToBlockTipSignal()
 {
     assert(!m_handler_notify_block_tip);
-    assert(!m_handler_notify_num_peers_changed);
 
     m_handler_notify_block_tip = m_node.handleNotifyBlockTip(
         [this](SynchronizationState state, interfaces::BlockTip tip, double verification_progress) {
             setBlockTipHeight(tip.block_height);
             setVerificationProgress(verification_progress);
         });
+}
 
-    m_handler_notify_num_peers_changed = m_node.handleNotifyNumConnectionsChanged(
-        [this](PeersNumByType new_num_peers) {
-            setNumOutboundPeers(new_num_peers.outbound_full_relay + new_num_peers.block_relay);
-        });
+void NodeModel::ConnectToNumConnectionsChangedSignal()
+{
+      assert(!m_handler_notify_num_peers_changed);
+
+      m_handler_notify_num_peers_changed = m_node.handleNotifyNumConnectionsChanged(
+          [this](PeersNumByType new_num_peers) {
+              setNumOutboundPeers(new_num_peers.outbound_full_relay + new_num_peers.block_relay);
+          });
 }
diff --git a/src/qml/nodemodel.h b/src/qml/nodemodel.h
index f81fce7a4..2cd3661f2 100644
--- a/src/qml/nodemodel.h
+++ b/src/qml/nodemodel.h
@@ -72,6 +72,7 @@ private:
     std::unique_ptr<interfaces::Handler> m_handler_notify_num_peers_changed;
 
     void ConnectToBlockTipSignal();
+    void ConnectToNumConnectionsChangedSignal();
 };
 
 #endif // BITCOIN_QML_NODEMODEL_H
```

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/127#issuecomment-1146895205).